modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPropertyDefValGenerator.cs

```diff
@@ -163,19 +164,81 @@ INamedTypeSymbol symbol
                     continue;
                 }
 
-                // TODO: Detect default value from simple property getters (currently we only detect from initializers)
-
-                EqualsValueClauseSyntax? initializer = property.DeclaringSyntaxReferences
-                    .Select(r => r.GetSyntax() as PropertyDeclarationSyntax)
-                    .Select(s => s?.Initializer ?? null)
-                    .FirstOrDefault();
+                var propertyDeclarationSyntax = property.DeclaringSyntaxReferences
+                    .Select(r => r.GetSyntax() as PropertyDeclarationSyntax).FirstOrDefault();
 
                 // Fully qualify the value to avoid issues with namespaces.
                 string? value = null;
-                if (initializer != null)
+                if (propertyDeclarationSyntax?.Initializer != null)
                 {
-                    var sm = context.Compilation.GetSemanticModel(initializer.SyntaxTree);
-                    value = initializer.Value.FullQualifiedSyntax(sm);
+                    var sm = context.Compilation.GetSemanticModel(propertyDeclarationSyntax.Initializer.SyntaxTree);
+                    value = propertyDeclarationSyntax.Initializer.Value.FullQualifiedSyntax(sm);
+                }
+
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    var propertyGet = propertyDeclarationSyntax?.AccessorList?.Accessors.Where(a => a.Keyword.IsKind(SyntaxKind.GetKeyword)).FirstOrDefault();
+                    if (propertyGet != null)
+                    {
+                        if (propertyGet.ExpressionBody != null)
+                        {
+                            if (propertyGet.ExpressionBody.Expression is IdentifierNameSyntax identifierNameSyntax)
+                            {
+                                var fieldSymbol = members
+                                    .Where(s => s.Kind == SymbolKind.Field)
+                                    .Cast<IFieldSymbol>()
+                                    .Where(s => s.Name == identifierNameSyntax.Identifier.Text)
+                                    .FirstOrDefault();
```

Comment:
This is likely error-prone. I think the following is the correct way to do it, but I can't confirm as I didn't try it:

```suggestion
                                var sm = context.Compilation.GetSemanticModel(identifierNameSyntax.SyntaxTree);

                                var fieldSymbol = sm.GetSymbolInfo(identifierNameSyntax).Symbol as IFieldSymbol;
```

Reply:
Both GetSemanticModel and Linq I tested without problems